### PR TITLE
Added support for suspend request

### LINKED ISSUE #489: Adding support for VM suspend request
Hello,

I am trying to schedule an automatic VM suspension. I have been following this tutorial [https://cloud.google.com/scheduler/docs/start-and-stop-compute-engine-instances-on-a-schedule#console](https://cloud.google.com/scheduler/docs/start-and-stop-compute-engine-instances-on-a-schedule#console
)

I am wondering if adding this feature is as easy as duplicating the "stop" code. I made a pull request with this change.

### PR DESCRIPTION
**Pull request by anaselfathi**
This is my first time looking at this code. I just copied the stop function and made modifications to post the suspend request.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-compute/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes  #489 🦕

### CODE REVIEW

**Comment by sofisl** [comment]
Hi @anaselfathi, thank you so much for contributing! 

So, admittedly, I'm not very familiar with this library, but I have some thoughts off the top of my head about your PR:

1) Would you mind writing a unit test in test/vm.js to cover the functionality?
2) If this code is the same as the 'stop' functionality, then what makes this code different/useful? 
3) It seems like you're creating a new endpoint here. I would, again, just confirm that it works (perhaps with some unit tests and general testing) as I'm not too familiar with the methods in this API.

Thanks in advance for your contribution!!

File: src/vm.js [hunk 1]
```diff
@@ -921,6 +921,45 @@ class VM extends common.ServiceObject {
       callback || common.util.noop
     );
   }
+  /**
+   * Suspend the instance.
+   *
+   * @see [Instances: Suspend API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instances/suspend}
```
Comment (sofisl) [inline src/vm.js:927]: this is probably the line that is making our docs tests angry. I think you could just add this link to the linkinator file to ignore for the time being.

**Comment by philipwigg** [comment]
I noticed that `suspend` is only in the beta API. - https://cloud.google.com/compute/docs/reference/rest/beta/instances/suspend

The v1 link in the PR actually 404s - https://cloud.google.com/compute/docs/reference/v1/instances/suspend

Not sure what the policy is for supporting beta features?

**Comment by anaselfathi** [comment]
Hello, thank you for taking the time to look at this.

I just saw that suspend is beta. In my opinion, I don't suggest to start adding support to beta functionalities.

For my particular issue (why I needed the suspend command). I just did a workaround where I actually use the "stop" command. But whenever I start the VM again I run a startup script (maybe this is even cleaner).

I am leaving a link to how to set up the startup script here: [](https://cloud.google.com/compute/docs/startupscript) in case someone else needs to do something similar.

Please disregard this PR.

Thank you again for looking at this issue.

**Comment by bcoe** [comment]
@anaselfathi thank you for the update 👌

I was wondering why I didn't see suspend in the docs. Let's revisit this when it comes out of beta?

Thanks for the work.

**Pull request closed without merging**